src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard-v3/dashboard/dashboard-v3.component.ts

```diff
@@ -92,6 +98,9 @@ export class DashboardV3Component extends PrometheusListHelper implements OnInit
     this.interval = this.refreshIntervalService.intervalData$.subscribe(() => {
       this.getHealth();
       this.getCapacityCardData();
+      this.getHardwareConfig().subscribe(() => {
+        this.getHardwareHealth();
+      });
```

Comment:
a better implementation would be to use the async pipes.

```
    this.hardwareSummary$ = this.hwSubject.pipe(
      switchMap(() => this.hardwareService.getHealth().pipe(
        switchMap((data: any) => {
          this.hasHardwareError = data.host.flawed;
          return of(data)
        })
      ))
    );
    this.isHardwareEnabled$ = this.getHardwareConfig();
    this.interval = this.refreshIntervalService.intervalData$.subscribe(() => {
         ........

      if (this.hardwareEnabled) this.hwSubject.next([]);
    });


  getHardwareConfig(): Observable<any> {
    return this.mgrModuleService.getConfig('cephadm').pipe(
      switchMap((resp: any) => {
        this.hardwareEnabled = resp?.hw_monitoring;
        return of(resp?.hw_monitoring)
      })
    );
  }
   
```
    
    and in html, 

```
    <ng-container *ngIf="isHardwareEnabled$ | async">
        <cd-card-row [data]="healthData.hosts"
                     link="/hosts"
                     title="Host"
                     summaryType="simplified"
                     *ngIf="healthData.hosts != null"
                     [dropdownEnabled]="hardwareEnabled"
                     [dropdownData]="hardwareSummary$ | async">
        </cd-card-row></ng-container>
```

Reply:
I had to make couple changes to your suggestion